### ISSUE TITLE
azurerm_postgressql_server requires ssl_enforcement_enabled but documentation says it's optional

### ISSUE THREAD

**Issue by everspader** [hashicorp/terraform-provider-azurerm#17877]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.5

### AzureRM Provider Version

3.16.0

### Affected Resource(s)/Data Source(s)

azurerm_postgresql_server

### Terraform Configuration Files

```hcl
data "azurerm_postgresql_server" "production_server" {
  name                = "prod_lite"
  resource_group_name = "server_prod"
}

resource "azurerm_resource_group" "rg" {
  name = "production_server"
  location = "northeurope"
}

resource "azurerm_postgresql_server" "postgres_server_backup" {
  name                = "prod_backup"
  location            = azurerm_resource_group.rg.location
  resource_group_name = azurerm_resource_group.rg.name

  sku_name = data.azurerm_postgresql_server.production_server.sku_name
  version  = data.azurerm_postgresql_server.production_server.version

  create_mode               = "PointInTimeRestore"
  restore_point_in_time     = "2022-08-04T06:00:00Z"
  creation_source_server_id = data.azurerm_postgresql_server.production_server.id
}
```

### Debug Output/Panic Output

```shell
Acquiring state lock. This may take a few moments...
╷
│ Error: Missing required argument
│ 
│   on main.tf line 13, in resource "azurerm_postgresql_server" "postgres_server_backup":
│   13: resource "azurerm_postgresql_server" "postgres_server_backup" {
│ 
│ The argument "ssl_enforcement_enabled" is required, but no definition was found.
```

### Expected Behaviour

Expected an output that matched the documentation. According to the original documentation, the argument `ssl_enforcement_enabled` should be optional when creating a new PostgreSQL server from scratch:

> [ssl_enforcement_enabled](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/postgresql_server#ssl_enforcement_enabled) - (Optional) Specifies if SSL should be enforced on connections. Possible values are true and false.

### Actual Behaviour

`terraform plan` requires the `ssl_enforcment_enabled` argument to be passed to the definition of the new postgres server:

### Steps to Reproduce

`terraform plan`

### References

[azurrm_postgresql_server](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/postgresql_server#ssl_enforcement_enabled)

**Comment by mikemadeja** [comment]
I'm a little knew to this, but I tested it with this update I did here and can deploy it w/o them on my build. I submitted a PR but I think I got it all.
https://github.com/hashicorp/terraform-provider-azurerm/pull/17889

**Comment by tombuildsstuff** [comment]
Fixed via #17889
